src/golibmc.go

```diff
@@ -73,23 +78,68 @@ var (
 	// Keys must be at maximum 250 bytes long, ASCII, and not
 	// contain whitespace or control characters.
 	ErrMalformedKey = errors.New("malformed: key is too long or contains invalid characters")
+
+	// ErrMemcachedClosed is returned when the memcached connection is
+	// already closing.
+	ErrMemcachedClosed = errors.New("memcached is closed")
+
+	// ErrBadConn is returned when the connection is out of its lifetime.
+	ErrBadConn = errors.New("bad conn")
+
+	errTemplate   = "libmc: network error(%s)"
+	badConnErrors []error
 )
 
+func init() {
+	badConnErrors = []error{
+		networkError(errorMessage[C.RET_MC_SERVER_ERR]),
+		networkError(errorMessage[C.RET_SEND_ERR]),
+		networkError(errorMessage[C.RET_CONN_POLL_ERR]),
+		networkError(errorMessage[C.RET_RECV_ERR]),
+	}
+}
+
 func networkError(msg string) error {
-	return errors.New("libmc: network error(" + msg + ")")
+	return fmt.Errorf(errTemplate, msg)
 }
 
-// Default memcached port
+// DefaultPort memcached port
 const DefaultPort = 11211
 
 // Client struct
 type Client struct {
-	_imp        unsafe.Pointer
-	servers     []string
-	prefix      string
-	noreply     bool
-	disableLock bool
-	lk          sync.Mutex
+	servers        []string
+	prefix         string
+	noreply        bool
+	disableLock    bool
+	hashFunc       int
+	failover       bool
+	connectTimeout C.int
+	pollTimeout    C.int
+	retryTimeout   C.int
+	lk             sync.Mutex
+	freeConns      []*conn
+	numOpen        int
```

Comment:
better expose some counters to watch the stat of the pool

Reply:
It is a difficult problem. If user update their values after I expose, the behavior of the connection pool can be crazy. For example, it seems that `database/sql` is not disclosed. How do you think about it?
https://github.com/golang/go/blob/master/src/database/sql/sql.go#L319-L347